packages/main/src/requests/request-helpers.ts

```diff
@@ -38,7 +39,52 @@ export function formatNewContent(
       }
     }
   }
-  return { role, parts: newParts };
+  // return { role, parts: newParts };
+  return assignRoleToPartsAndValidateSendMessageRequest(newParts);
+}
+
+/**
+ * When multiple Part types (i.e. FunctionResponsePart and TextPart) are
+ * passed in a single Part array, we may need to assign different roles to each
+ * part. Currently only FunctionResponsePart requires a role other than 'user'.
+ * @private
+ * @param parts Array of parts to pass to the model
+ * @returns Array of content items
+ */
+function assignRoleToPartsAndValidateSendMessageRequest(
```

Comment:
I think this (through formatNewContent) may end up being called in the ChatSession constructor when an existing history is passed in which includes `model` turns, so it needs to preserve the `model` role when that happens.

Reply:
Yeah, you are right. I'm just not sure if `startChat` method should accept anything but `Content` with roles properly assigned.

I would assume the use case for `startChat` with history is only valid if the history is obtained through `getHistory` method (or constructed as a proper Content).

In that case we may drop `InputContent` interface at all. WDYT?

Reply:
Added new validation function for chat history